packages/syft/src/syft/service/network/node_peer.py

```diff
@@ -155,8 +229,109 @@ def proxy_from(self, client: SyftClient) -> SyftClient:
+        highest_priority_route: NodeRoute = self.node_routes[-1]
         for route in self.node_routes:
-            if route.priority > final_route.priority:
-                final_route = route
-        return final_route
+            if route.priority > highest_priority_route.priority:
+                highest_priority_route = route
+        return highest_priority_route
+
+    def delete_route(
+        self, route: NodeRouteType | None = None, route_id: UID | None = None
+    ) -> SyftError | None:
+        """
+        Deletes a route from the peer's route list.
+        Takes O(n) where is n is the number of routes in self.node_routes.
+
+        Args:
+            route (NodeRouteType): The route to be deleted;
+            route_id (UID): The id of the route to be deleted;
+
+        Returns:
+            SyftError: If deleting failed
+        """
+        if route_id:
+            try:
+                self.node_routes = [r for r in self.node_routes if r.id != route_id]
+            except Exception as e:
+                return SyftError(
+                    message=f"Error deleting route with id {route_id}. Exception: {e}"
+                )
+
+        if route:
+            try:
+                same_route: Callable = _route_type_to_same_route_check(route)
+                self.node_routes = [
+                    r for r in self.node_routes if not same_route(r, route)
+                ]
+            except Exception as e:
+                return SyftError(
+                    message=f"Error deleting route with id {route.id}. Exception: {e}"
+                )
+
+        return None
+
+
+def _route_type_to_same_route_check(
+    route: NodeRouteType,
+) -> Callable[[NodeRouteType, NodeRouteType], bool]:
+    """
+    Takes a route as input and returns a function that can be
+    used to compare if the two routes are the same.
+
+    Args:
+        route (NodeRouteType): The route for which to get a comparison function.
+
+    Returns:
+        Callable[[NodeRouteType, NodeRouteType], bool]: A function that takes two routes as input and returns a boolean
+        indicating whether the routes are the same.
+    """
+    route_type_to_comparison_method: dict[
+        type[NodeRouteType], Callable[[NodeRouteType, NodeRouteType], bool]
+    ] = {
+        HTTPNodeRoute: _same_http_route,
+        PythonNodeRoute: _same_python_route,
+        VeilidNodeRoute: _same_veilid_route,
+    }
+    return route_type_to_comparison_method[type(route)]
+
+
+def _same_http_route(route: HTTPNodeRoute, other: HTTPNodeRoute) -> bool:
+    """
+    Check if two HTTPNodeRoute are the same based on protocol, host_or_ip (url) and port
+    """
+    if type(route) != type(other):
+        return False
+    return (
+        (route.host_or_ip == other.host_or_ip)
+        and (route.port == other.port)
+        and (route.protocol == other.protocol)
+    )
+
+
+def _same_python_route(route: PythonNodeRoute, other: PythonNodeRoute) -> bool:
```

Comment:
Similarly for PythonNodeRoute define `__eq__` method and move this logic there

Reply:
Done @shubham3121